seaborn/regression.py

```diff
@@ -127,8 +127,13 @@ def __init__(self, x, y, data=None, x_estimator=None, x_bins=None,
         else:
             self.x_discrete = self.x
 
+        # Disable regression in case of singleton inputs
+        if len(self.x) <= 1:
+            self.fit_reg = False
```

Comment:
Perhaps warn here if fit_reg was initially set to True (current default)?

Reply:
I considered that but I am generally not a fan of warnings being raised as a consequence of default arguments. In most cases, I would not think people would expect to draw a regression line for singleton inputs. (I think the main situation you'll encounter this code path is in a faceted context, where the warning would be somewhat confusing.)